v3/go.mod

```diff
@@ -0,0 +1,5 @@
+module github.com/sendgrid/rest/v3
```

Comment:
I don't think it's a good idea to have a different folders. We can simply replace the version in the original `go.mod` file instead. Please revert these changes in the `v3` folder.

Reply:
Even though the https://go.dev/blog/v2-go-modules recommends having different folders, that's for projects that are actually introducing breaking changes, and this is not the case in this PR.

Reply:
@marcoshuck Thanks for your review. If you take a look at [these failed jobs](https://github.com/zwartho/rest/actions/runs/5753835843), breaking changes are seemingly apparent when referencing the `io` module. This [SO question](https://stackoverflow.com/questions/75206234/for-go-ioutil-readall-ioutil-readfile-ioutil-readdir-deprecated) explains it nicely, but essentially the only Go changes made in this MR were replacing references to the deprecated `ioutil` module. I could not get tests for Go 1.14 and 1.15 passing without utilizing the v3 directory and splitting up the test job into two separate jobs. So in order to prevent breaking functionality for users of Go 1.14 and 1.15, I believe the need for the v3 directory and separate test jobs is necessary. If you have any suggestions to mitigate this issue in a different way, please let me know and I can accommodate.

Reply:
That's a very a good point. We need to have this discussion with some of the members of the @sendgrid team, my 2c: We should replace `ioutil` with `io` package references or any other alternative that makes it work. Bumping to a new major version should allow us to deprecate previous Go versions.

Reply:
@twilio-dx @SendGridDX Any suggested next steps?